bindings/wasm/tests/storage.ts

```diff
@@ -186,58 +186,132 @@ describe("#JwkStorageDocument", function() {
+            type: "UniversityDegreeCredential",
+            credentialSubject: {
+                id: holderDoc.id(),
+                degree: {
+                    type: "BachelorDegree",
+                    name: "Bachelor of Science and Arts",
+                },
+            },
+            issuer: issuerDoc.id(),
+            issuanceDate: Timestamp.nowUTC(),
+        };
 
         const credential = new Credential(credentialFields);
-        // Create the JWT
-        const credentialJwt = await doc.createCredentialJwt(storage, fragment, credential, new JwsSignatureOptions());
+        const credentialJwt: Jwt = await issuerDoc.createCredentialJwt(
+            storage,
+            fragment,
+            credential,
+            new JwsSignatureOptions(),
+        );
 
-        // Check that the credentialJwt can be decoded and verified
-        let credentialValidator = new JwtCredentialValidator();
-        const credentialRetrieved = credentialValidator.validate(
-            credentialJwt,
-            doc,
-            JwtCredentialValidationOptions.default(),
+        const presentation = new JwtPresentation({
+            holder: holderDoc.id(),
+            verifiableCredential: [credentialJwt.toString(), credentialJwt.toString()],
+        });
+
+        const expirationDate = Timestamp.nowUTC().checkedAdd(Duration.days(2));
+        const audience = "did:test:123";
+        const presentationJwt = await holderDoc.createPresentationJwt(
+            storage,
+            fragment,
+            presentation,
+            new JwsSignatureOptions(),
+            new JwtPresentationOptions({
+                expirationDate,
+                issuanceDate: Timestamp.nowUTC(),
+                audience,
+            }),
+        );
+
+        let validator = new JwtPresentationValidator(customVerifier);
+        let decoded: DecodedJwtPresentation = validator.validate(
+            presentationJwt,
+            holderDoc,
+            [issuerDoc],
+            JwtPresentationValidationOptions.default(),
             FailFast.FirstError,
-        ).credential();
-        assert.deepStrictEqual(credentialRetrieved.toJSON(), credential.toJSON());
+        );
 
-        // Also check using our custom verifier
-        let credentialValidatorCustom = new JwtCredentialValidator(customVerifier);
-        const credentialRetrievedCustom = credentialValidatorCustom.validate(
-            credentialJwt,
-            doc,
-            JwtCredentialValidationOptions.default(),
-            FailFast.AllErrors,
-        ).credential();
-        // Check that customVerifer.verify was indeed called
-        assert.deepStrictEqual(customVerifier.verifications(), 2);
-        assert.deepStrictEqual(credentialRetrievedCustom.toJSON(), credential.toJSON());
+        assert.deepStrictEqual(decoded.credentials()[0].credential().toJSON(), credential.toJSON());
+        assert.equal(decoded.expirationDate()?.toString(), expirationDate?.toString());
```

Comment:
```suggestion
        assert.equal(decoded.expirationDate()!.toString(), expirationDate!.toString());
```

We know those properties are set, so we shouldn't use `?`, right? If they don't end up being set, we would like the test to fail.